### DISCUSSION TITLE
I want to create a page that filters my table using two Option Pickers

### DISCUSSION THREAD

**Discussion by iwanhoogendoorn** [Budibase/budibase#11148] [Help] [answered]
Hi, I have a table with A Buch of records.
I want to use the **first** option picker that takes the list data from a column and filters out the countries.

I then want to have a **second** options picker with A, B, C, D, etc. and these are static values.
When the "A" is selected the table needs to be filtered again and only show me a selection of Cities that start with an "A".

How can this be done?

**Comment by R2bEEaton** [comment, marked as answer]
Hello, this could be done using Dynamic Filters or a Form.

### Dynamic Filter

[Dynamic Filters](https://docs.budibase.com/docs/dynamic-filter) are useful for allowing end users to filter data based on their own methods. For example, if a Dynamic Filter was set up like this:

![image](https://github.com/Budibase/budibase/assets/34921506/4fd6eafb-77aa-4138-a309-894f7cf80384)

The end user could then filter it based on country name and city starting letter like this:

![image](https://github.com/Budibase/budibase/assets/34921506/fc575279-c11b-4c9c-a30b-16e5c203f21d)
![image](https://github.com/Budibase/budibase/assets/34921506/1163f548-7ff0-4671-a3e5-972a5a8881c9)

The benefit of this is that it allows users to come up with their own filtering methods. For example, you could filter by only the first letter of the city. Or get all cities from United States or Greece. The drawback however is that there is no dropdown for the list of countries that you wanted. To achieve this, it would be possible but less efficient with forms.

### Form

If Dynamic Filters aren't what you want, you can do it with a form with an option picker populated by the country column of a data source. Then you could have a text box or other option picker with the alphabet. Then, a search button would redirect you to a new page with some URL parameters that would then filter the data. I have attached a rough example that you can important explore.

[nxninja-export-1688657199625.tar.gz](https://github.com/Budibase/budibase/files/11969824/nxninja-export-1688657199625.tar.gz)

### Hope this helps!
Let me know if you have any more questions. Otherwise, mark this question as answered to help future readers! 😃

**Reply by JCV67** [reply]
Hello, you can do this by adding a filter to the data provider and than adding the option picker field  between the data provider and the table. If you're using table block, you must eject it first.

**Reply by R2bEEaton** [reply]
Yeah, that's similar to what I did before I changed it to the search button. Doing it like you described works but it's annoying because the data for the options picker is filtered by itself. That means that if you choose Greece from the dropdown, and go back into the dropdown, only Greece will be available because it just filtered itself. You could do two data providers, but this would not be ideal with a lot of rows compounded with the fact that there are also row limits. I like the dynamic filter option most, even though it's not exactly what the original poster was asking for.
